esrally/utils/versions.py

```diff
@@ -180,6 +184,8 @@ def best_match(available_alternatives, distribution_version):
         major, _, _, _ = components(distribution_version)
         if major > _latest_major(available_alternatives):
             return "master"
+    elif is_serverless(distribution_version):
+        return "master"
```

Comment:
We only every select the `master` branch of `rally-teams` when targeting a serverless cluster.